src/openms/source/FORMAT/DATAACCESS/SiriusMzTabWriter.cpp

```diff
@@ -43,10 +43,33 @@
 using namespace OpenMS;
 using namespace std;
 
-int SiriusMzTabWriter::extract_scan_index(const String &path)
+int SiriusMzTabWriter::extractScanIndex(const String& path)
 {
   boost::regex regexp_ind("--(?<SCAN>\\d+)--");
-  return SpectrumLookup::extractScanNumber(path, regexp_ind, false);;
+  return SpectrumLookup::extractScanNumber(path, regexp_ind, false);
+}
+
+int SiriusMzTabWriter::extractScanNumber(const String& path)
+{
+  boost::regex regexp("-(?<SCAN>\\d+)--");
+  return SpectrumLookup::extractScanNumber(path, regexp, false);
+}
+
+String SiriusMzTabWriter::extractFeatureId(const String& path)
+{
+  boost::smatch match;
+  String feature_id;
+  boost::regex regexp_feature("_(?<SCAN>\\d+)-");
+  bool found = boost::regex_search(path, match, regexp_feature);
+  if (found && match["SCAN"].matched) 
+  {
+    feature_id = "id_" + match["SCAN"].str();
+  }
+  if (feature_id.empty() || feature_id == "id_0")
+  {
+    feature_id = "null";
+  }
```

Comment:
are all cases covered?

Reply:
I think so - in any other case there should be a feature id